src/commands/apphosting-rollouts-create.ts

```diff
@@ -1,27 +1,116 @@
+  .option("-c, --commit <commit>", "git commit to deploy (defaults to the latest commit)", "")
+  .option("-f, --force", "skip confirmation")
   .before(apphosting.ensureApiEnabled)
-  .action(async (backendId: string, buildId: string, options: Options) => {
+  .action(async (backendId: string, options: Options) => {
     const projectId = needProjectId(options);
     const location = options.location as string;
-    // TODO: Should we just reuse the buildId?
-    const rolloutId =
-      (options.buildId as string) ||
-      (await apphosting.getNextRolloutId(projectId, location, backendId));
-    const build = `projects/${projectId}/backends/${backendId}/builds/${buildId}`;
-    const op = await apphosting.createRollout(projectId, location, backendId, rolloutId, {
-      build,
+
+    let branch = options.gitBranch as string | undefined;
+    const commit = options.commit as string | undefined;
+    if (branch && commit) {
+      throw new FirebaseError(
+        "Cannot specify both a branch and commit to deploy. Please specify either --git-branch or --commit.",
+      );
+    }
+    const backend = await apphosting.getBackend(projectId, location, backendId);
+    if (!backend.codebase.repository) {
+      throw new FirebaseError(
+        `Failed to find connected Git repository for backend ${backendId}. Please check that your backend has been successfully linked to a Git repository.`,
+      );
+    }
+    const { repoLink, owner, repo, readToken } = await getRepoDetailsFromBackend(
+      projectId,
+      location,
+      backend.codebase.repository,
+    );
+
+    let target;
+    if (branch) {
+      const branches = await listAllBranches(repoLink.name);
+      if (!branches.has(branch)) {
+        throw new FirebaseError(
+          `Unrecognized git branch ${branch}. Please double-check your branch name and try again.`,
+        );
+      }
+      const branchInfo = await getGitHubBranch(owner, repo, branch, readToken.token);
+      target = branchInfo.commit;
+    } else if (commit) {
+      if (!/^(?:[0-9a-f]{40}|[0-9a-f]{7})$/.test(commit)) {
+        throw new FirebaseError(`Invalid git commit ${commit}. Must be a valid SHA1 hash.`);
+      }
+      try {
+        const commitInfo = await getGitHubCommit(owner, repo, commit, readToken.token);
+        target = commitInfo;
+      } catch (err: unknown) {
+        if ((err as FirebaseError).status === 422) {
+          throw new FirebaseError(
+            `Unrecognized git commit ${commit}. Please double-check your commit hash and try again.`,
+          );
+        }
+        throw err;
+      }
+    } else {
+      branch = await promptGitHubBranch(repoLink);
+      const branchInfo = await getGitHubBranch(owner, repo, branch, readToken.token);
+      target = branchInfo.commit;
+    }
+
+    logBullet(`You are about to deploy [${target.sha.substring(0, 7)}]: ${target.commit.message}`);
```

Comment:
let's make sure we update these as well

Reply:
What should we make sure to update here?